diff --git a/dist/plugin/utils/ast-utils.js b/dist/plugin/utils/ast-utils.js
index 19a8dca48f757af2afead625f026c77fbf94778d..cb887244cca0781a3a0f211b575ab429f534e8fc 100644
--- a/dist/plugin/utils/ast-utils.js
+++ b/dist/plugin/utils/ast-utils.js
@@ -64,8 +64,8 @@ exports.createBooleanLiteral = createBooleanLiteral;
 exports.createPrimitiveLiteral = createPrimitiveLiteral;
 exports.createLiteralFromAnyValue = createLiteralFromAnyValue;
 const tsdoc_1 = require("@microsoft/tsdoc");
-const ts = __importStar(require("typescript"));
-const typescript_1 = require("typescript");
+const ts = __importStar(require("typescript-legacy"));
+const typescript_1 = require("typescript-legacy");
 const plugin_utils_1 = require("./plugin-utils");
 function renderDocNode(docNode) {
     let result = '';
diff --git a/dist/plugin/utils/external-imports.util.js b/dist/plugin/utils/external-imports.util.js
index 04f3f0424c9315ae7e4d72352fb2c9197b3bbb28..052e0145a8d2dbcccc2bf0a2142f79122fd4ba3a 100644
--- a/dist/plugin/utils/external-imports.util.js
+++ b/dist/plugin/utils/external-imports.util.js
@@ -2,7 +2,7 @@
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.getExternalImports = getExternalImports;
 exports.replaceExternalImportsInTypeReference = replaceExternalImportsInTypeReference;
-const ts = require("typescript");
+const ts = require("typescript-legacy");
 function getExternalImports(sourceFile) {
     const externalImports = {};
     const importDeclarations = sourceFile.statements.filter(ts.isImportDeclaration);
diff --git a/dist/plugin/utils/plugin-utils.js b/dist/plugin/utils/plugin-utils.js
index 2ffb8953b0b3ed38d6e8243d64fb7c505eef5f70..52b22169a11661a6d86e4b7e7f19d80a4c6f5b2d 100644
--- a/dist/plugin/utils/plugin-utils.js
+++ b/dist/plugin/utils/plugin-utils.js
@@ -53,7 +53,7 @@ exports.normalizePackagePath = normalizePackagePath;
 exports.canReferenceNode = canReferenceNode;
 const lodash_1 = require("lodash");
 const path_1 = require("path");
-const ts = __importStar(require("typescript"));
+const ts = __importStar(require("typescript-legacy"));
 const ast_utils_1 = require("./ast-utils");
 function getDecoratorOrUndefinedByNames(names, decorators, factory) {
     return (decorators || factory.createNodeArray()).find((item) => {
diff --git a/dist/plugin/utils/type-reference-to-identifier.util.js b/dist/plugin/utils/type-reference-to-identifier.util.js
index c4aeee78ba8e455db3c699189c1e004156ab870a..7199f2fe9c40360021001909d6d4f15562dde678 100644
--- a/dist/plugin/utils/type-reference-to-identifier.util.js
+++ b/dist/plugin/utils/type-reference-to-identifier.util.js
@@ -35,7 +35,7 @@ var __importStar = (this && this.__importStar) || (function () {
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.typeReferenceToIdentifier = typeReferenceToIdentifier;
 const path_1 = require("path");
-const ts = __importStar(require("typescript"));
+const ts = __importStar(require("typescript-legacy"));
 const plugin_debug_logger_1 = require("../plugin-debug-logger");
 const plugin_utils_1 = require("./plugin-utils");
 function typeReferenceToIdentifier(typeReferenceDescriptor, hostFilename, options, factory, type, typeImports) {
diff --git a/dist/plugin/visitors/abstract.visitor.js b/dist/plugin/visitors/abstract.visitor.js
index d1ace213a38b305039552cd6b804ea7051252238..0289520e98522f016b5450bbf94d51bc0a6deb37 100644
--- a/dist/plugin/visitors/abstract.visitor.js
+++ b/dist/plugin/visitors/abstract.visitor.js
@@ -34,7 +34,7 @@ var __importStar = (this && this.__importStar) || (function () {
 })();
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.AbstractFileVisitor = void 0;
-const ts = __importStar(require("typescript"));
+const ts = __importStar(require("typescript-legacy"));
 const plugin_constants_1 = require("../plugin-constants");
 const [major, minor] = ts.versionMajorMinor.split('.').map((x) => +x);
 class AbstractFileVisitor {
diff --git a/dist/plugin/visitors/controller-class.visitor.js b/dist/plugin/visitors/controller-class.visitor.js
index 98eb7fff8336bc6dc7d59baa02a1dde7a7cad12f..6fdb0f8c19df5d6c64c2f79c0ec8be496adf56c0 100644
--- a/dist/plugin/visitors/controller-class.visitor.js
+++ b/dist/plugin/visitors/controller-class.visitor.js
@@ -37,7 +37,7 @@ exports.ControllerClassVisitor = void 0;
 const common_1 = require("@nestjs/common");
 const lodash_1 = require("lodash");
 const path_1 = require("path");
-const ts = __importStar(require("typescript"));
+const ts = __importStar(require("typescript-legacy"));
 const decorators_1 = require("../../decorators");
 const plugin_constants_1 = require("../plugin-constants");
 const ast_utils_1 = require("../utils/ast-utils");
diff --git a/dist/plugin/visitors/model-class.visitor.js b/dist/plugin/visitors/model-class.visitor.js
index 8b45c9ef54614e757dae0ac89c5004ce5c5d1e20..d3f03e9c43a1dea9d2cf53ddff373fb3af1bbe72 100644
--- a/dist/plugin/visitors/model-class.visitor.js
+++ b/dist/plugin/visitors/model-class.visitor.js
@@ -36,8 +36,8 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.ModelClassVisitor = void 0;
 const lodash_1 = require("lodash");
 const path_1 = require("path");
-const ts = __importStar(require("typescript"));
-const typescript_1 = require("typescript");
+const ts = __importStar(require("typescript-legacy"));
+const typescript_1 = require("typescript-legacy");
 const decorators_1 = require("../../decorators");
 const decorators_properties_1 = require("../../services/decorators-properties");
 const plugin_constants_1 = require("../plugin-constants");
diff --git a/dist/plugin/visitors/readonly.visitor.js b/dist/plugin/visitors/readonly.visitor.js
index d3f68ac8ba654ce6ad3271e42045d4ac00f6fef5..3b1f6b3e7ff4d47357ac84118c9fa850ce6d5daa 100644
--- a/dist/plugin/visitors/readonly.visitor.js
+++ b/dist/plugin/visitors/readonly.visitor.js
@@ -34,7 +34,7 @@ var __importStar = (this && this.__importStar) || (function () {
 })();
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.ReadonlyVisitor = void 0;
-const ts = __importStar(require("typescript"));
+const ts = __importStar(require("typescript-legacy"));
 const merge_options_1 = require("../merge-options");
 const is_filename_matched_util_1 = require("../utils/is-filename-matched.util");
 const controller_class_visitor_1 = require("./controller-class.visitor");
