diff --git a/dist/plugin/compiler-plugin.d.ts b/dist/plugin/compiler-plugin.d.ts
index be0604abe3588f705c4db02790fe6952b71e3c23..0da65ec5b6aa0670b0c0077242b029abedd2db08 100644
--- a/dist/plugin/compiler-plugin.d.ts
+++ b/dist/plugin/compiler-plugin.d.ts
@@ -1,2 +1,2 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 export declare const before: (options?: Record<string, any>, program?: ts.Program) => (ctx: ts.TransformationContext) => ts.Transformer<any>;
diff --git a/dist/plugin/index.d.ts b/dist/plugin/index.d.ts
index edf94fdbac15521def568fb001d5ec5a0de8b075..f40a984e9518f592a737f1eecb0324bd587649ab 100644
--- a/dist/plugin/index.d.ts
+++ b/dist/plugin/index.d.ts
@@ -1,4 +1,4 @@
-import type * as ts from 'typescript';
+import type * as ts from 'typescript-legacy';
 export * from './compiler-plugin.js';
 export * from './visitors/readonly.visitor.js';
 declare const _default: (program: ts.Program, options?: Record<string, any>) => (ctx: ts.TransformationContext) => ts.Transformer<any>;
diff --git a/dist/plugin/utils/ast-utils.d.ts b/dist/plugin/utils/ast-utils.d.ts
index 519f246474b46d79253a6fe8bdb552e9a81522a8..396df7f81ad6e7aa2c297b62293f3344263211c0 100644
--- a/dist/plugin/utils/ast-utils.d.ts
+++ b/dist/plugin/utils/ast-utils.d.ts
@@ -1,6 +1,6 @@
 import { DocComment, DocNode } from '@microsoft/tsdoc';
-import * as ts from 'typescript';
-import { Decorator, Node, ObjectFlags, Type, TypeChecker, TypeFlags, TypeFormatFlags, UnionTypeNode } from 'typescript';
+import * as ts from 'typescript-legacy';
+import { Decorator, Node, ObjectFlags, Type, TypeChecker, TypeFlags, TypeFormatFlags, UnionTypeNode } from 'typescript-legacy';
 export declare function renderDocNode(docNode: DocNode): string;
 export declare function isArray(type: Type): boolean;
 export declare function getTypeArguments(type: Type): any;
diff --git a/dist/plugin/utils/ast-utils.js b/dist/plugin/utils/ast-utils.js
index e922a08f6a22481508826161037eba8ad0e7a5f6..d0aaa2b3ba49eb7266f021dde4cff84f5caf6a5a 100644
--- a/dist/plugin/utils/ast-utils.js
+++ b/dist/plugin/utils/ast-utils.js
@@ -1,6 +1,6 @@
 import { DocExcerpt, TSDocParser } from '@microsoft/tsdoc';
-import * as ts from 'typescript';
-import { ObjectFlags, SyntaxKind, TypeFlags, TypeFormatFlags } from 'typescript';
+import * as ts from 'typescript-legacy';
+import { ObjectFlags, SyntaxKind, TypeFlags, TypeFormatFlags } from 'typescript-legacy';
 import { isDynamicallyAdded } from './plugin-utils.js';
 export function renderDocNode(docNode) {
     let result = '';
diff --git a/dist/plugin/utils/module-format.util.d.ts b/dist/plugin/utils/module-format.util.d.ts
index a57970a8e8a2d132886c5af32612769b8930bf59..18301201ae48a5e5993465ccebb64ca3c1d0d83d 100644
--- a/dist/plugin/utils/module-format.util.d.ts
+++ b/dist/plugin/utils/module-format.util.d.ts
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { PluginOptions } from '../merge-options.js';
 export declare function resolvePluginOptionsForFile(options: PluginOptions, sourceFile: ts.SourceFile, compilerOptions: ts.CompilerOptions): PluginOptions;
 export declare function isEsmOutputFile(sourceFile: ts.SourceFile, compilerOptions: ts.CompilerOptions, options?: PluginOptions): boolean;
diff --git a/dist/plugin/utils/module-format.util.js b/dist/plugin/utils/module-format.util.js
index 0113db9f81479ab5a3bab1094929e4eff179ce0f..76742e935b7725bc81d5f0104b013f88501179ad 100644
--- a/dist/plugin/utils/module-format.util.js
+++ b/dist/plugin/utils/module-format.util.js
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 export function resolvePluginOptionsForFile(options, sourceFile, compilerOptions) {
     return {
         ...options,
diff --git a/dist/plugin/utils/plugin-utils.d.ts b/dist/plugin/utils/plugin-utils.d.ts
index fc29262f3d4803bc38f65b01892ae239e6aca7ba..d8fea8685bfef6e5f6ca369ea42ba5726dadce2e 100644
--- a/dist/plugin/utils/plugin-utils.d.ts
+++ b/dist/plugin/utils/plugin-utils.d.ts
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { PluginOptions } from '../merge-options.js';
 export declare function getDecoratorOrUndefinedByNames(names: string[], decorators: readonly ts.Decorator[], factory: ts.NodeFactory): ts.Decorator | undefined;
 export declare function getTypeReferenceAsString(type: ts.Type, typeChecker: ts.TypeChecker, arrayDepth?: number): {
diff --git a/dist/plugin/utils/plugin-utils.js b/dist/plugin/utils/plugin-utils.js
index 631c6996e83a9e3da12c573313c0a168e34b3c33..a641ea4ddc48b6b8730c18dd50f801464d39bc90 100644
--- a/dist/plugin/utils/plugin-utils.js
+++ b/dist/plugin/utils/plugin-utils.js
@@ -1,7 +1,7 @@
 import { head } from 'es-toolkit/compat';
 import { createRequire } from 'node:module';
 import { isAbsolute, posix } from 'path';
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { getDecoratorName, getText, getTypeArguments, hasFlag, isArray, isBigInt, isBoolean, isEnum, isInterface, isNumber, isString, isStringLiteral, isStringMapping } from './ast-utils.js';
 const moduleRequire = createRequire(import.meta.url);
 export function getDecoratorOrUndefinedByNames(names, decorators, factory) {
diff --git a/dist/plugin/utils/type-reference-to-identifier.util.d.ts b/dist/plugin/utils/type-reference-to-identifier.util.d.ts
index 665582c634676a40ee5cbbba89fee7c6cc46fdca..6c3bfc3b3ee4635a250014744ad4100f1efc75df 100644
--- a/dist/plugin/utils/type-reference-to-identifier.util.d.ts
+++ b/dist/plugin/utils/type-reference-to-identifier.util.d.ts
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { PluginOptions } from '../merge-options.js';
 export declare function typeReferenceToIdentifier(typeReferenceDescriptor: {
     typeName: string;
diff --git a/dist/plugin/utils/type-reference-to-identifier.util.js b/dist/plugin/utils/type-reference-to-identifier.util.js
index 59a2524133062aa502a93c5f17918509ef89ccf5..b6413ffcc996b78429cdc90471254ae0df4af761 100644
--- a/dist/plugin/utils/type-reference-to-identifier.util.js
+++ b/dist/plugin/utils/type-reference-to-identifier.util.js
@@ -1,5 +1,5 @@
 import { posix } from 'path';
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { pluginDebugLogger } from '../plugin-debug-logger.js';
 import { convertPath, extractTypeArgumentIfArray, getOutputExtension, replaceImportPath } from './plugin-utils.js';
 export function typeReferenceToIdentifier(typeReferenceDescriptor, hostFilename, options, factory, type, typeImports, hoistedTypeImports) {
diff --git a/dist/plugin/visitors/abstract.visitor.d.ts b/dist/plugin/visitors/abstract.visitor.d.ts
index c6fb1c434443ffc4bfb7d5f14ab468df0897a09b..1bddc81c478d76039a1ac470d178caa76e8493d4 100644
--- a/dist/plugin/visitors/abstract.visitor.d.ts
+++ b/dist/plugin/visitors/abstract.visitor.d.ts
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { PluginOptions } from '../merge-options.js';
 export declare class AbstractFileVisitor {
     protected readonly _fileOutputExtensions: Record<string, string>;
diff --git a/dist/plugin/visitors/abstract.visitor.js b/dist/plugin/visitors/abstract.visitor.js
index a353838275ea6a521f8e5696ce8cb9308256af5d..3f9b344491246fa7d1a1113411bf884f7e8c8945 100644
--- a/dist/plugin/visitors/abstract.visitor.js
+++ b/dist/plugin/visitors/abstract.visitor.js
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { OPENAPI_NAMESPACE, OPENAPI_PACKAGE_NAME } from '../plugin-constants.js';
 import { isEsmOutputFile } from '../utils/module-format.util.js';
 import { getOutputExtension, normalizePackagePath } from '../utils/plugin-utils.js';
diff --git a/dist/plugin/visitors/controller-class.visitor.d.ts b/dist/plugin/visitors/controller-class.visitor.d.ts
index 9f343adb54828c191163a906082d91229ee5453a..85aada97ecf1a732768b00a2631d5bda3b384ef1 100644
--- a/dist/plugin/visitors/controller-class.visitor.d.ts
+++ b/dist/plugin/visitors/controller-class.visitor.d.ts
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { PluginOptions } from '../merge-options.js';
 import { AbstractFileVisitor } from './abstract.visitor.js';
 type ClassMetadata = Record<string, ts.ObjectLiteralExpression>;
diff --git a/dist/plugin/visitors/controller-class.visitor.js b/dist/plugin/visitors/controller-class.visitor.js
index 3b1af54d3adbaefb1550b063f1f1df625f065dd6..38f9ff26404529e6b6dac1330238168204083851 100644
--- a/dist/plugin/visitors/controller-class.visitor.js
+++ b/dist/plugin/visitors/controller-class.visitor.js
@@ -1,7 +1,7 @@
 import { HttpStatus } from '@nestjs/common';
 import { compact, head } from 'es-toolkit/compat';
 import { posix } from 'path';
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { ApiOperation, ApiParam, ApiQuery, ApiResponse } from '../../decorators/index.js';
 import { OPENAPI_NAMESPACE } from '../plugin-constants.js';
 import { collectExistingApiParamNames, createLiteralFromAnyValue, getDecoratorArguments, getDecoratorName, getJSDocParamDescriptionsOfNode, getMainCommentOfNode, getNamedParamDecoratorArg, getTsDocErrorsOfNode, getTsDocTagsOfNode } from '../utils/ast-utils.js';
diff --git a/dist/plugin/visitors/model-class.visitor.d.ts b/dist/plugin/visitors/model-class.visitor.d.ts
index 4305f6ca9f176e3b790ababebbecf0bc3d41c374..2e46b8417a35f470082e9a9b055c1a4517a4d139 100644
--- a/dist/plugin/visitors/model-class.visitor.d.ts
+++ b/dist/plugin/visitors/model-class.visitor.d.ts
@@ -1,5 +1,5 @@
-import * as ts from 'typescript';
-import { PropertyAssignment } from 'typescript';
+import * as ts from 'typescript-legacy';
+import { PropertyAssignment } from 'typescript-legacy';
 import { PluginOptions } from '../merge-options.js';
 import { AbstractFileVisitor } from './abstract.visitor.js';
 type ClassMetadata = Record<string, ts.ObjectLiteralExpression>;
diff --git a/dist/plugin/visitors/model-class.visitor.js b/dist/plugin/visitors/model-class.visitor.js
index 506767b7ef35efef8a07c4ff3dec262fdbbfcf6e..b0188c03851f7c4c0e24b1e55fb7522ee6a42075 100644
--- a/dist/plugin/visitors/model-class.visitor.js
+++ b/dist/plugin/visitors/model-class.visitor.js
@@ -1,7 +1,7 @@
 import { compact, flatten, head } from 'es-toolkit/compat';
 import { posix } from 'path';
-import * as ts from 'typescript';
-import { factory } from 'typescript';
+import * as ts from 'typescript-legacy';
+import { factory } from 'typescript-legacy';
 import { ApiHideProperty, ApiProperty } from '../../decorators/index.js';
 import { decoratorsProperties, decoratorsPropertiesMappingType } from '../../services/decorators-properties.js';
 import { METADATA_FACTORY_NAME } from '../plugin-constants.js';
diff --git a/dist/plugin/visitors/readonly.visitor.d.ts b/dist/plugin/visitors/readonly.visitor.d.ts
index b82c92a62e971670c6e3898c8d3d5441ab99d17a..64e4dc132163dc1dddbd72b68d1fb1e3c447fb27 100644
--- a/dist/plugin/visitors/readonly.visitor.d.ts
+++ b/dist/plugin/visitors/readonly.visitor.d.ts
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { PluginOptions } from '../merge-options.js';
 export declare class ReadonlyVisitor {
     private readonly options;
diff --git a/dist/plugin/visitors/readonly.visitor.js b/dist/plugin/visitors/readonly.visitor.js
index 815105b78bad89c35546a948ae33320a4632d1dd..3f0ff6d2dac0aa2aee247f488ee8f3fb44b8c809 100644
--- a/dist/plugin/visitors/readonly.visitor.js
+++ b/dist/plugin/visitors/readonly.visitor.js
@@ -1,4 +1,4 @@
-import * as ts from 'typescript';
+import * as ts from 'typescript-legacy';
 import { mergePluginOptions } from '../merge-options.js';
 import { isFilenameMatched } from '../utils/is-filename-matched.util.js';
 import { ControllerClassVisitor } from './controller-class.visitor.js';
