diff --git a/dist/esm/lib/verify-typescript-setup.js b/dist/esm/lib/verify-typescript-setup.js
index cea10266321489b1ee40c1b5f902baa1627e142f..365b69b3052ad4e6fe8acdbb0b673d6b188f3d79 100644
--- a/dist/esm/lib/verify-typescript-setup.js
+++ b/dist/esm/lib/verify-typescript-setup.js
@@ -12,7 +12,7 @@ import { isCI } from '../server/ci-info';
 import { missingDepsError } from './typescript/missingDependencyError';
 import { resolveFrom } from './resolve-from';
 const typescriptPackage = {
-    file: 'typescript/lib/typescript.js',
+    file: 'typescript/package.json',
     pkg: 'typescript',
     exportsRestrict: true
 };
@@ -132,7 +132,7 @@ export async function verifyAndRunTypeScript({ dir, distDir, cacheDir, strictRou
             // Install native bindings so that code frame rendering works in the worker
             const { installBindings } = require('../build/swc/install-bindings');
             await installBindings();
-            const tsPath = deps.resolved.get('typescript');
+            const tsPath = resolveFrom(dir, 'typescript');
             const typescript = await Promise.resolve(require(tsPath));
             // Verify the project passes type-checking before we go to webpack phase:
             result = await runTypeCheck(typescript, dir, distDir, resolvedTsConfigPath, cacheDir, hasAppDir, {
diff --git a/dist/lib/verify-typescript-setup.js b/dist/lib/verify-typescript-setup.js
index caa1e6ad592c663e1392fe1e393e398e5aef19ad..2a407a1b2a862a6dd68166f555d5bd392c92d27e 100644
--- a/dist/lib/verify-typescript-setup.js
+++ b/dist/lib/verify-typescript-setup.js
@@ -68,7 +68,7 @@ function _interop_require_wildcard(obj, nodeInterop) {
     return newObj;
 }
 const typescriptPackage = {
-    file: 'typescript/lib/typescript.js',
+    file: 'typescript/package.json',
     pkg: 'typescript',
     exportsRestrict: true
 };
@@ -188,7 +188,7 @@ async function verifyAndRunTypeScript({ dir, distDir, cacheDir, strictRouteTypes
             // Install native bindings so that code frame rendering works in the worker
             const { installBindings } = require('../build/swc/install-bindings');
             await installBindings();
-            const tsPath = deps.resolved.get('typescript');
+            const tsPath = (0, _resolvefrom.resolveFrom)(dir, 'typescript');
             const typescript = await Promise.resolve(require(tsPath));
             // Verify the project passes type-checking before we go to webpack phase:
             result = await runTypeCheck(typescript, dir, distDir, resolvedTsConfigPath, cacheDir, hasAppDir, {
